client/txn_test.go

```diff
@@ -288,7 +288,7 @@ func TestEndWriteRestartReadOnlyTransaction(t *testing.T) {
 			t.Errorf("expected error: %t, got error: %v", !success, err)
 		}
 		if !reflect.DeepEqual(expCalls, calls) {
-			t.Fatalf("expected %v, got %v", expCalls, calls)
+			t.Errorf("expected %v, got %v", expCalls, calls)
```

Comment:
No, you're not missing anything. But why bother printing the second iteration? There is a failure which will get a human's attention. That seems sufficient.